dagmc/dagnav.py

```diff
@@ -18,19 +18,30 @@ def __init__(self, moab_file):
             self.mb = core.Core()
             self.mb.load_file(moab_file)
 
+        self.used_ids = {}
+        self.used_ids[Surface._geom_dimension] = set(self.surfaces_by_id.keys())
+        self.used_ids[Volume._geom_dimension] = set(self.volumes_by_id.keys())
+        self.used_ids[Group._geom_dimension] = set(group.id for group in self.groups.values())
+
     def _sets_by_category(self, set_type : str):
         """Return all sets of a given type"""
         return self.mb.get_entities_by_type_and_tag(self.mb.get_root_set(), types.MBENTITYSET, [self.category_tag], [set_type])
 
     @property
     def surfaces(self):
-        surfaces = [Surface(self, h) for h in self._sets_by_category('Surface')]
-        return {s.id: s for s in surfaces}
+        return [Surface(self, h) for h in self._sets_by_category('Surface')]
+    
+    @property
+    def surfaces_by_id(self):
+        return {s.id: s for s in self.surfaces}
 
     @property
     def volumes(self):
-        volumes = [Volume(self, h) for h in self._sets_by_category('Volume')]
-        return {v.id: v for v in volumes}
+        return [Volume(self, h) for h in self._sets_by_category('Volume')]
+
+    @property
+    def volumes_by_id(self):
+        return {v.id: v for v in self.volumes}
 
     @property
     def groups(self) -> Dict[str, Group]:
```

Comment:
Along the lines of the discussion in #12, to be consistent with the `volumes` and `surfaces` properties this should probably return a list of `Group` instances and we can add a another method called `groups_by_name` to get the name --> `Group` instance mapping.